File: backend/python/common/topics.py
"""Redpanda topic names. Partitioned by symbol; replayable; idempotent."""

TICKS_RAW = "ticks-raw"
TICKS_NORMALIZED = "ticks-normalized"

CANDLES = {
    "1m": "candles-1m",
    "5m": "candles-5m",
    "15m": "candles-15m",
    "1h": "candles-1h",
    "1d": "candles-1d",
}

OPTIONS_CHAIN_SNAPSHOTS = "options-chain-snapshots"
SIGNALS = "signals"
PAPER_TRADE_EVENTS = "paper-trade-events"
NEWS_RAW = "news-raw"
NEWS_PROCESSED = "news-processed"
REGIME_EVENTS = "regime-events"

File: backend/rust/crates/gx-core/src/topics.rs
//! Direct port of `backend/python/common/topics.py`.

pub const TICKS_RAW: &str = "ticks-raw";
pub const TICKS_NORMALIZED: &str = "ticks-normalized";
pub const OPTIONS_CHAIN_SNAPSHOTS: &str = "options-chain-snapshots";
pub const SIGNALS: &str = "signals";
pub const PAPER_TRADE_EVENTS: &str = "paper-trade-events";
pub const NEWS_RAW: &str = "news-raw";
pub const NEWS_PROCESSED: &str = "news-processed";
pub const REGIME_EVENTS: &str = "regime-events";

pub const CANDLE_INTERVALS: &[&str] = &["1m", "5m", "15m", "1h", "1d"];

pub fn candles(interval: &str) -> String {
    format!("candles-{interval}")
}

pub fn candles_topic(interval: &str) -> &'static str {
    match interval {
        "1m" => "candles-1m",
        "5m" => "candles-5m",
        "15m" => "candles-15m",
        "1h" => "candles-1h",
        "1d" => "candles-1d",
        _ => "candles-1m",
    }
}
